### ISSUE TITLE
Porting PyTorch layer norm to Flax

### ISSUE THREAD

**Issue by gadgetsam** [google/flax#2195]
I'm trying to port the layer norm module from PyTorch to Flax. I transformed the state dict from PyTorch to flax yet the layer norm is still not producing the same results. I tried this on OSX 12.4 on M1, Ubuntu 20.04 and google colab and in all of them the outputs aren't equal. 

I created a minimal example: 
`import flax
import torch
import jax.numpy as jnp
import numpy as np

def test_layernorm():
    torch_layernorm = torch.nn.LayerNorm(12)
    flax_layernorm = flax.linen.LayerNorm(12)
    torch_state_dict = torch_layernorm.state_dict()
    torch_state_dict["scale"] = jnp.array(np.array(torch_state_dict.pop("weight")))
    torch_state_dict["bias"] = jnp.array(np.array(torch_state_dict.pop("bias")))

    x = torch.randn((8, 12))
    x_flax = jnp.array(np.array(x))

    torch_out = torch_layernorm(x)
    flax_out = flax_layernorm.apply(variables={"params": torch_state_dict}, x=x_flax)
    assert np.allclose(torch_out.detach(), flax_out)

test_layernorm()`

And created a colab link: https://colab.research.google.com/drive/1wTIbWbM9LBjzlKC14aegLjWHrkoXdapF#scrollTo=hv0tS0BJ2_EE 

Thanks in advance for the help.

**Comment by marcvanzee** [comment]
I don't think you can expect these to be the same since the weights are initialized with different seed. You should instead copy the weights from Flax to Pytorch after initializing (or the other way around). Then the results should be the same.

**Comment by marcvanzee** [comment]
Flax uses shape inference, so you do not have to provide the normalized shape as an input to the constructor for `LayerNorm`. Because you do provide an argument (`12`), this means `epsilon` will be set to 12. The following code should work:

```py
import torch
import jax.numpy as jnp
import numpy as np


def test_layernorm():
    torch_layernorm = torch.nn.LayerNorm(12)
    flax_layernorm = flax.linen.LayerNorm()
    torch_state_dict = torch_layernorm.state_dict()
    torch_state_dict["scale"] = jnp.array(np.array(torch_state_dict.pop("weight")))
    torch_state_dict["bias"] = jnp.array(np.array(torch_state_dict.pop("bias")))

    x = torch.randn((8, 12))
    x_flax = jnp.array(np.array(x))

    torch_out = torch_layernorm(x)
    flax_out = flax_layernorm.apply(variables={"params": torch_state_dict}, x=x_flax)
    assert np.allclose(torch_out.detach(), flax_out)


test_layernorm()
```
